### ISSUE TITLE
Constructors parameters must be on their own line and have a trailing comma

### ISSUE THREAD

**Issue by DaveLiddament** [PHP-CS-Fixer/PHP-CS-Fixer#6325]
## Rule request

For PHP8+ it'd be nice to enforce that constructors parameters are on their own line and they have a trailing comma. 

Similar to #6079, but rather than line length it is applied if the constructor has 1 or more parameters.

### Code snippet that shows how the rule would fix PHP code

```diff
-  public function __construct(private PersonRespository $personRepository, private CostCalculator $costCalculator) {
+  public function __construct(
+    private PersonRespository $personRepository, 
+    private CostCalculator $costCalculator, 
+  ) {
```

## Thanks

By the way, thanks for a great tool. We use it on all our projects and it really helps us.

**Comment by kubawerlos** [comment]
Is this because of promoted properties? If so take a look at [`MultilinePromotedPropertiesFixer`](https://github.com/kubawerlos/php-cs-fixer-custom-fixers#MultilinePromotedPropertiesFixer).

Then you have to update [`TrailingCommaInMultilineFixer`](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.6.0/doc/rules/control_structure/trailing_comma_in_multiline.rst) configuration by adding `parameters`.

**Comment by DaveLiddament** [comment]
@kubawerlos thanks for link. Yes it looks like those rules should help me out. 

Initially my use case is for promoted properties, but thinking about this more and given trailing commas are allowed on all parameters I should probably apply this rule to all my methods. I'll take a look at the  linked Fixer. I'm sure I can make alter it for my needs.

**Comment by gharlan** [comment]
> Is this because of promoted properties? If so take a look at [`MultilinePromotedPropertiesFixer`](https://github.com/kubawerlos/php-cs-fixer-custom-fixers#MultilinePromotedPropertiesFixer).

👍 I would love to see this fixer here in the official repo.

**Comment by short-cut** [comment]
Yes, this would be very nice!

**Comment by RichLinds1988** [comment]
+1 on this

**Comment by ibousfiha** [comment]
+1 !
This will improve readability!

**Comment by horlyk** [comment]
> 👍 I would love to see this fixer here in the official repo.

Vote for it too!

**Comment by wannevancamp** [comment]
+1 on this one

**Comment by short-cut** [comment]
Yes, would be very nice, I do it nearly everywhere when I write my constructors.

**Comment by Wirone** [comment]
Thank you all for your interest in this feature request and your feedback. But I would like to ask everyone to use 👍 reaction instead of commenting "+1" to minimise amount of notifications that all participants (and maintainers) receive. Also it makes it easier to [sort issues by reaction](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc). But feel free to comment if you want to provide some information that would be helpful for implementation or rise some concerns to discuss.

Thanks in advance 🙂.

**Comment by fridtjof** [comment]
Still interested in and watching this. :)

**Comment by keradus** [comment]
closed by #8595
